python/hsfs/storage_connector.py

```diff
@@ -1244,14 +1248,29 @@ def read(
         # Returns
             `Dataframe`: A Spark dataframe.
         """
-
+        if not engine.get_instance().is_connector_type_supported(
+            StorageConnector.BIGQUERY
+        ):
```

Comment:
Maybe this is cleaner
```suggestion
        if not engine.get_instance().is_connector_type_supported(self.type):
```